docs/src/prototypes/mentions/MentionsEditor.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react'
+import * as _ from 'lodash'
+import keyboardKey from 'keyboard-key'
+
+import { atMentionItems, AtMentionItem } from './dataMocks'
+import { insertTextAtCursorPosition } from './utils'
+import { PortalAtCursorPosition } from './PortalAtCursorPosition'
+
+export interface MentionsContainerProps {
+  items?: AtMentionItem[]
+  searchQuery?: string
+  open?: boolean
+  onInputKeyDown?: (e: React.KeyboardEvent) => void
+  onOpenChange?: (e: React.SyntheticEvent, data: { open?: boolean }) => void
+  onSearchQueryChange?: (e: React.SyntheticEvent, data: { searchQuery: string }) => void
+  onSelectedChange?: (e: React.SyntheticEvent, data: { searchQuery: string }) => void
+}
+
+interface MentionsEditorState {
+  open?: boolean
+  searchQuery?: string
+}
+
+const editorStyle: React.CSSProperties = {
+  backgroundColor: '#eee',
+  borderRadius: '5px',
+  border: '1px dashed grey',
+  padding: '5px',
+  minHeight: '100px',
+  outline: 0,
+}
+
+const withMentionsEditor = <P extends MentionsContainerProps = MentionsContainerProps>(
```

Comment:
Please consider to avoid the HOC pattern and especially combine it with hooks.

You can just use `MentionsDropdown` as component, currently I don't see how it can be reused.

Reply:
It can be reused for instance to create the same prototype with Popup: https://github.com/stardust-ui/react/compare/proto/mentions-with-popup?expand=1

@layershifter What is the alternative to HOC pattern if I still want to avoid coupling the editor and the @mentions dropdown?

Reply:
You can use renderProps pattern in this case:
```
<MentionsEditor>
  {editorProps => <Component {...editorProps} />
</MentionsEditor>
```

Reply:
done, thanks